web/modules/custom/dpl_related_content/dpl_related_content.module

```diff
@@ -0,0 +1,292 @@
+<?php
+
+use Drupal\Core\Database\Database;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Entity\FieldableEntityInterface;
+use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Drupal\recurring_events\Entity\EventInstance;
+
+/**
+ * Get terms from field as an array of IDs.
+ *
+ * @return array<int>
+ *   Term IDs.
+ */
+function _dpl_related_content_get_term_ids(FieldableEntityInterface $entity, string $field_name): array {
```

Comment:
This has a lot of "private" functions at the top of the module" while the meat, `dpl_related_content_preprocess_page()`, is at the bottom.  That makes it a bit harder to read as you start with the details but do not know where you will end.